tests/test_jobqueue.py

```diff
@@ -325,3 +326,29 @@ def test_context_based_callback(self, job_queue):
         sleep(0.03)
 
         assert self.result == 0
+
+    def test_job_next_t_property(self, job_queue):
+        # Testing:
+        # - next_t values match values from self._queue.queue (for run_once and run_repeating jobs)
+        # - next_t equals None if job is removed or if it's already ran
+
+        job1 = job_queue.run_once(self.job_run_once, 0.05, name='run_once job')
+        job2 = job_queue.run_once(self.job_run_once, 0.05, name='canceled run_once job')
+        job_queue.run_repeating(self.job_run_once, 0.02, name='repeatable job')
+
+        sleep(0.03)
+        job2.schedule_removal()
+
+        with job_queue._queue.mutex:
+            for t, job in job_queue._queue.queue:
+                t = dtm.datetime.fromtimestamp(t, job.tzinfo)
+
+                if job.removed:
+                    assert job.next_t is None
+                else:
+                    assert job.next_t == t
+
+        sleep(0.03)
+
+        assert job1.next_t is None
+        assert job2.next_t is None
```

Comment:
Could you add assertions checking that `self.result` (updated by `self.job_run_once`) has the correct value after each job execution? Just for completenes sake …

Reply:
Added.

Reply:
I also noticed that setters Job.days and Job.interval raise ValueError when TypeError is more appropriate - changed it to TypeError in 6b1fce5 and 8231df1